pkg/s3wrapper/filesystem.go

```diff
@@ -0,0 +1,184 @@
+package s3wrapper
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	logutil "github.com/openshift/assisted-service/pkg/log"
+
+	"github.com/moby/moby/pkg/ioutils"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type FSClient struct {
+	log     *logrus.Logger
+	basedir string
+}
+
+func NewFSClient(basedir string, logger *logrus.Logger) *FSClient {
+	return &FSClient{log: logger, basedir: basedir}
+}
+
+func (c *FSClient) IsAwsS3() bool {
+	return false
+}
+
+func (f *FSClient) CreateBucket() error {
+	return nil
+}
+
+func (f *FSClient) Upload(ctx context.Context, data []byte, objectName string) error {
+	log := logutil.FromContext(ctx, f.log)
+	filePath := filepath.Join(f.basedir, objectName)
+	if err := os.MkdirAll(path.Dir(filePath), 0755); err != nil {
```

Comment:
it will fail if directory already exists? why not to validate it?

Reply:
It's fine if it already exists.  It's like "mkdir -p" - it creates any directories that are missing in the path.